macros/blueprint-proc-macro/src/tangle/mod.rs

```diff
@@ -3,72 +3,84 @@ use quote::{format_ident, quote};
 use syn::Ident;
 
 #[allow(clippy::too_many_lines)]
-pub fn field_type_to_param_token(ident: &Ident, t: &FieldType) -> proc_macro2::TokenStream {
+pub fn field_type_to_param_token(
+    ident: &Ident,
+    t: &FieldType,
+    panic_on_decode_fail: bool,
+) -> proc_macro2::TokenStream {
+    let else_block = if panic_on_decode_fail {
+        quote! {
+            panic!("Failed to decode the field");
+        }
+    } else {
+        quote! {
+            return Ok(vec![]);
+        }
+    };
     match t {
         FieldType::Void => unreachable!("void type should not be in params"),
         FieldType::Bool => {
-            quote! { let Some(Field::Bool(#ident)) = args_iter.next() else { return Ok(vec![]); }; }
+            quote! { let Some(gadget_sdk::tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::field::Field::Bool(#ident)) = args_iter.next() else { #else_block }; }
```

Comment:
We'll need some way to make this switch between two types when we have this on mainnet as well. Would be good to identify what that could look like sooner than later.